app/src/main/java/com/group80/uoftinder/UserScoreFacade.java

```diff
@@ -0,0 +1,87 @@
+package com.group80.uoftinder;
+
+import com.group80.uoftinder.entities.User;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class UserScoreFacade {  // using Facade Design Principle to delegate tasks
+    private final UserScoreCalculator usCalc;
+    private final UserScoreComparator usComp;
+    private ArrayList<HashSet<Integer>> userAnswers; // [{0, 2, 3}, {1, 3}, ...]
+    private final int[] answerSchema;
+    private final boolean[] isMultiSelect;
+    private final int answerLen;
+
+    public UserScoreFacade(User currentUser, CreateAccountInteractor cai) {  // using Dependency Injection Design Principle
+        this.userAnswers = currentUser.getAnswers();
```

Comment:
For here, I suggest that the second param, cai, is unnecessary. I believe that you should be able to find the type of user through the currentUser param and then retrieve the set of questions based on the type of user.

Reply:
Not really sure how the group wants to fix this problem at the moment. Might take some further discussion to resolve how we want to store/retrieve this data.

Reply:
Glad to see how we resolved this issue after communicating in person to reach an agreement that it's better to design `getAnswerSchema` and  `getIsMultiSelect` as static methods in `CreateAccountInteractor` since the return values do not differ by instances of the object!

Reply:
Yep, should be resolved!

Reply:
Thank you for considering my suggestion! I believe that the code is much cleaner now as the UserScoreFacade now requires 1 param instead of 2.